pkg/kubelet/kubelet.go

```diff
@@ -2043,6 +2054,13 @@ func (kl *Kubelet) getPodsToSync() []*api.Pod {
 	}
 	var podsToSync []*api.Pod
 	for _, pod := range allPods {
+		for _, podSyncLoopObserver := range kl.ObservablePodSyncLooper {
+			if podSyncLoopObserver.ShouldSync(pod) {
+				podsToSync = append(podsToSync, pod)
+				break
```

Comment:
well, i said that wrong, it should be a `break`, but we need to move the pod uid checks up first so we dont get duplicate entries.